### ISSUE TITLE
running evosuite-1.0.3.jar

### ISSUE THREAD

**Issue by JimAthrs** [EvoSuite/evosuite#105]
### Context
to do my daily work, i need to be on our customer vpn. However when I'm on that vpn and run "java -jar evosuite-1.0.3.jar -base_dir ./scratch -seed 1000 -prefix my.package.name.prefix -projectCP REAL_CLASSPATH" I get the following error: 

[MASTER] 12:12:19.752 [logback-2] ERROR ClientServices - Failed to register client services
java.rmi.ConnectIOException: Exception creating connection to: 172.16.3.84; nested exception is: 
	java.net.SocketException: Permission denied: connect
	at sun.rmi.transport.tcp.TCPEndpoint.newSocket(TCPEndpoint.java:631) ~[na:1.8.0_121]
	at sun.rmi.transport.tcp.TCPChannel.createConnection(TCPChannel.java:216) ~[na:1.8.0_121]
	at sun.rmi.transport.tcp.TCPChannel.newConnection(TCPChannel.java:202) ~[na:1.8.0_121]
	at sun.rmi.server.UnicastRef.newCall(UnicastRef.java:342) ~[na:1.8.0_121]
	at sun.rmi.registry.RegistryImpl_Stub.rebind(Unknown Source) ~[na:1.8.0_121]
	at org.evosuite.rmi.ClientServices.registerServices(ClientServices.java:66) ~[evosuite-1.0.3.jar:1.0.3]
	at org.evosuite.ClientProcess.run(ClientProcess.java:82) [evosuite-1.0.3.jar:1.0.3]
	at org.evosuite.ClientProcess.main(ClientProcess.java:202) [evosuite-1.0.3.jar:1.0.3]
Caused by: java.net.SocketException: Permission denied: connect
	at java.net.DualStackPlainSocketImpl.connect0(Native Method) ~[na:1.8.0_121]
	at java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:79) ~[na:1.8.0_121]
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350) ~[na:1.8.0_121]
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206) ~[na:1.8.0_121]
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188) ~[na:1.8.0_121]
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172) ~[na:1.8.0_121]
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392) ~[na:1.8.0_121]
	at java.net.Socket.connect(Socket.java:589) ~[na:1.8.0_121]
	at java.net.Socket.connect(Socket.java:538) ~[na:1.8.0_121]
	at java.net.Socket.<init>(Socket.java:434) ~[na:1.8.0_121]
	at java.net.Socket.<init>(Socket.java:211) ~[na:1.8.0_121]
	at sun.rmi.transport.proxy.RMIDirectSocketFactory.createSocket(RMIDirectSocketFactory.java:40) ~[na:1.8.0_121]
	at sun.rmi.transport.proxy.RMIMasterSocketFactory.createSocket(RMIMasterSocketFactory.java:148) ~[na:1.8.0_121]
	at sun.rmi.transport.tcp.TCPEndpoint.newSocket(TCPEndpoint.java:613) ~[na:1.8.0_121]
	... 7 common frames omitted
[MASTER] 12:12:19.774 [logback-2] ERROR ClientProcess - Error when generating tests for: my.package.name.prefix.ConfigurableHandlerResolver with seed 1000. Configuration id : null
java.lang.RuntimeException: Could not connect to master process on port 10262
	at org.evosuite.ClientProcess.run(ClientProcess.java:87) ~[evosuite-1.0.3.jar:1.0.3]
	at org.evosuite.ClientProcess.main(ClientProcess.java:202) ~[evosuite-1.0.3.jar:1.0.3]

When I'm not on the customer vpn, your tool works and generates the testsuite. I even tried tying that IP address to localhost in Windows 7 hosts file and had no luck. I even tried starting your tool and then connect to the customer vpn and same error results. 

Are there any work-arounds? What about changing the EVOsuite source locally and recompiling (which we've not ever done)?

### Steps to Reproduce
hopefully the above is what you need.

### EvoSuite Arguments
hopefully the above is what you need.

### Current Result
hopefully the above is what you need. 
 
### Expected result
hopefully the above is what you need.

### Additional info
*Please add any information of interest here below*

**Comment by gofraser** [comment]
By default, RMI tries to connect to localhost (i.e. 127.0.0.1). I don't know much about Windows, but is it possible that you don't have localhost set up to the loopback device when the VPN is running? If localhost does exist, a shot in the dark: can you try "set _JAVA_OPTIONS=java.rmi.server.hostname=127.0.0.1" or "set _JAVA_OPTIONS=java.rmi.server.hostname=localhost" before running EvoSuite?

**Comment by JimAthrs** [comment]
Hello Gordon,

No luck: still the same issue. What is w/ the IP address of 172.16.3.14 anyways? It has been awhile I’ve done RMI myself.


Thanks...Jim

Jim Snyder
Senior Software Engineer, Configuration Manager, VBMS Awards
CHS / Critical Networks / HARRIS CORPORATION
Office: 321-355-2018
harris.com<http://www.harris.com/>  | james.snyder@harris.com<mailto:nsharp02@harris.com>

**Comment by gofraser** [comment]
I assume that IP address is either your address, or the one of the virtual device used for the VPN. Can you ping 127.0.0.1? Can you ping localhost? (What does it resolve to?)

**Comment by JimAthrs** [comment]
Thank you for the quick response.

Yes the IP address is what the VPN is assigning to me. Doing an ipconfig /all showed it.

The ping’ing of 127.0.0.1 and localhost pointed to my machine name.

Not sure where to go from here.


Thanks...Jim

Jim Snyder
Senior Software Engineer, Configuration Manager, VBMS Awards
CHS / Critical Networks / HARRIS CORPORATION
Office: 321-355-2018
harris.com<http://www.harris.com/>  | james.snyder@harris.com<mailto:nsharp02@harris.com>

**Comment by gofraser** [comment]
Another quick thing to check for: Can you try if that problem goes away if you deactivate any Windows firewalls running on the computer?

**Comment by JimAthrs** [comment]
Unfortunately I can’t deactivate due to employer’s policy.

I did let EVOsuite run while at lunch and it generated a lot of ESTest*.java and I’m trying to get those to run during our mvn build. I’m having trouble get the evosuite jars to download via our pom.xml entries (per your http://www.evosuite.org/documentation/maven-plugin/ site).

Basically getting

[ERROR] Failed to execute goal on project my.project: Could not resolve dependencies for project a18k.awards:my.project:jar:13.0.134: Failed to collect dependencies at org.evosuite:evosuite-standalone-runtime:jar:1.0.3: Failed to read artifact descriptor for org.evosuite:evosuite-standalone-runtime:jar:1.0.3: Could not transfer artifact org.evosuite:evosuite-standalone-runtime:pom:1.0.3 from/to shibboleth (http://svn.shibboleth.net/view/java-parent-project/trunk/): svn.shibboleth.net: Unknown host svn.shibboleth.net -> [Help 1]

Thanks...Jim

Jim Snyder
Senior Software Engineer, Configuration Manager, VBMS Awards
CHS / Critical Networks / HARRIS CORPORATION
Office: 321-355-2018
harris.com<http://www.harris.com/>  | james.snyder@harris.com<mailto:nsharp02@harris.com>

**Comment by JimAthrs** [comment]
Nevermind I forgot to include both pluginRepositories and EvoSuite <repository> entries


Thanks...Jim

Jim Snyder
Senior Software Engineer, Configuration Manager, VBMS Awards
CHS / Critical Networks / HARRIS CORPORATION
Office: 321-355-2018
harris.com<http://www.harris.com/>  | james.snyder@harris.com<mailto:nsharp02@harris.com>




From: Snyder, James (U.S. Person)
Sent: Friday, April 21, 2017 3:38 PM
To: 'EvoSuite/evosuite' <reply@reply.github.com>; EvoSuite/evosuite <evosuite@noreply.github.com>
Cc: Author <author@noreply.github.com>
Subject: RE: [Marketing] Re: [EvoSuite/evosuite] running evosuite-1.0.3.jar (#105)

Unfortunately I can’t deactivate due to employer’s policy.

I did let EVOsuite run while at lunch and it generated a lot of ESTest*.java and I’m trying to get those to run during our mvn build. I’m having trouble get the evosuite jars to download via our pom.xml entries (per your http://www.evosuite.org/documentation/maven-plugin/ site).

Basically getting

[ERROR] Failed to execute goal on project my.project: Could not resolve dependencies for project a18k.awards:my.project:jar:13.0.134: Failed to collect dependencies at org.evosuite:evosuite-standalone-runtime:jar:1.0.3: Failed to read artifact descriptor for org.evosuite:evosuite-standalone-runtime:jar:1.0.3: Could not transfer artifact org.evosuite:evosuite-standalone-runtime:pom:1.0.3 from/to shibboleth (http://svn.shibboleth.net/view/java-parent-project/trunk/): svn.shibboleth.net: Unknown host svn.shibboleth.net -> [Help 1]

Thanks...Jim

Jim Snyder
Senior Software Engineer, Configuration Manager, VBMS Awards
CHS / Critical Networks / HARRIS CORPORATION
Office: 321-355-2018
harris.com<http://www.harris.com/>  | james.snyder@harris.com<mailto:nsharp02@harris.com>

**Comment by gofraser** [comment]
I don't know how to reproduce this problem so I don't really see what I could do about it. I understand that there is the personal firewall policy, but if you can't even deactivate it briefly, is there some way you can see in that personal firewall if anything is being blocked? This seems the most likely explanation currently.
